### ISSUE TITLE
crash during game

### ISSUE THREAD

**Issue by askovpen** [L-EARN/NoCheatZ-4#57]
```
[NoCheatZ v4-beta-r1-156-gc9968c1-dirty-debug] DEBUG : CNoCheatZPlugin::ClientDisconnect
Dropped sasai from server: Kicked by Console : [NoCheatZ 4] ConVar request timed out
L 07/04/2016 - 13:57:45: [csgo_country_v2.smx] 2 round end, 1:1(3) - 9 - $2000$
[STEAM_1:1:129722145] Pure server: file: GAME\sound/myarena/bombtimer_girl/6sec.mp3 ( 1 20479 4fff ) could not open file to hash ( benign for now ) : 870f6ad26e4de6333958ee3a0977233b : 

[NoCheatZ v4-beta-r1-156-gc9968c1-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE51D060 -> player, spec_mode 3)
[STEAM_1:1:129722145] Pure server: file: GAME\sound/myarena/bombtimer_girl/2sec.mp3 ( 1 16299 3fab ) could not open file to hash ( benign for now ) : ef1d495ceff13565f3b01afcbd1b3c17 : 

[NoCheatZ v4-beta-r1-156-gc9968c1-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE51D060 -> player, spec_mode 3)
[NoCheatZ v4-beta-r1-156-gc9968c1-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE51D060 -> player, spec_scoreboard )
[STEAM_1:1:129722145] Pure server: file: GAME\sound/levels_ranks/levelup.mp3 ( 1 18002 4652 ) could not open file to hash ( benign for now ) : 02e7fd2910fcc4f065a530343e1f206a : 

[NoCheatZ v4-beta-r1-156-gc9968c1-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE51D060 -> player, spec_mode 4)

Program received signal SIGSEGV, Segmentation fault.
0xec7ea2fc in NczPlayer::GetIndex (this=0x1ba0) at ../server-plugin/Code/Players/NczPlayer.h:85
85              return m_index;
(gdb) bt
#0  0xec7ea2fc in NczPlayer::GetIndex (this=0x1ba0) at ../server-plugin/Code/Players/NczPlayer.h:85
#1  0xec7ff07d in VisCache::IsValid (this=0x9006480, pa=0xae41c50, pb=0x1ba0)
    at Code/Systems/Blockers/WallhackBlocker.h:104
#2  0xec7fd8e2 in WallhackBlocker::RT_SetTransmitCallback (this=0x90037f0, sender_player=..., receiver_player=...)
    at Code/Systems/Blockers/WallhackBlocker.cpp:156
#3  0xec7f22f9 in SetTransmitHookListener::RT_nSetTransmit (This=0xaba5ee0, pInfo=0xeeba928, bAlways=false)
    at Code/Hooks/SetTransmitHookListener.cpp:70
#4  0xf2f81ee0 in CServerGameEnts::CheckTransmit(CCheckTransmitInfo*, unsigned short const*, int) ()
   from /mnt/p/home/steam/csgo/csgo/bin/server.so
#5  0xf69b771b in SV_ComputeClientPacks(int, CGameClient**, CFrameSnapshot*) ()
   from /mnt/p/home/steam/csgo/bin/engine.so
#6  0xf69b402e in CGameServer::SendClientMessages(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#7  0xf69b43b1 in SV_SendClientUpdates(bool, bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#8  0xf69b44e0 in SV_Frame(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#9  0xf692308e in _Host_RunFrame_Server(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#10 0xf6923d2c in _Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#11 0xf6924468 in Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#12 0xf692d714 in CHostState::State_Run(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#13 0xf692d9e6 in CHostState::FrameUpdate(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#14 0xf692daf9 in HostState_Frame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#15 0xf69ca66a in CEngine::Frame() () from /mnt/p/home/steam/csgo/bin/engine.so
#16 0xf69c75f6 in CDedicatedServerAPI::RunFrame() () from /mnt/p/home/steam/csgo/bin/engine.so
#17 0xf784cc2d in RunServerIteration(bool) () from ./bin/dedicated.so
#18 0xf784ccb8 in RunServer(bool) () from ./bin/dedicated.so
#19 0xf69c76ed in CModAppSystemGroup::Main() () from /mnt/p/home/steam/csgo/bin/engine.so
#20 0xf6b17e70 in CAppSystemGroup::Run() () from /mnt/p/home/steam/csgo/bin/engine.so
#21 0xf69c88b7 in CDedicatedServerAPI::ModInit(ModInfo_t&) () from /mnt/p/home/steam/csgo/bin/engine.so
#22 0xf784cd94 in CDedicatedAppSystemGroup::Main() () from ./bin/dedicated.so
#23 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#24 0xf787196e in CSteamApplication::Main() () from ./bin/dedicated.so
#25 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#26 0xf7812d73 in main () from ./bin/dedicated.so
#27 0x080486f9 in main ()
```
ps: i kicked on first line, but false positive.

**Comment by L-EARN** [comment]
I was not able to reproduce the crash.
I tested spectating bots in eye but everything worked fine.

Judging by the value of the this pointer, maybe there is an overflow somewhere ...

If you are able to reproduce it, please tell me how.

**Comment by L-EARN** [comment]
Fixed in e4a61a9
